src/Api/Endpoints/Candidate/Visit/Image/Format/Raw.php

```diff
@@ -0,0 +1,172 @@
+    public function __construct(\LORIS\Image $image)
+    {
+        $this->_image = $image;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return array('GET');
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return array(
+                'v0.0.2',
+                'v0.0.3-dev',
+               );
+    }
+
+    /**
+     * Delegates the request to a private handler appropriate to the requets's
+     * method.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+
+        case 'OPTIONS':
+            return (new \LORIS\Http\Response())
+                ->withHeader('Allow', $this->allowedMethods());
+
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+
+        }
+    }
+
+    /**
+     * Create an array representation of this endpoint's reponse body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        $info = $this->_image->getFileInfo();
+
+        if (!$info->isFile()) {
+            error_log('file in database but not in file system');
+            return new \LORIS\Http\Response\NotFound();
+        }
+
+        if (!$info->isReadable()) {
+            error_log('file exist but is not readable by webserver');
+            return new \LORIS\Http\Response\NotFound();
+        }
+
+        $mincpath = \NDB_Factory::singleton()->config()
+            ->getSetting('MINCToolsPath');
+
+        $minctooldir = new \SPLFileInfo($mincpath);
+        if (!$minctooldir->isDir() || !$minctooldir->isReadable()) {
+            return new \LORIS\Http\Response\InternalServerError(
+                'Invalid MINCToolsPath configuration setting.'
+            );
+        }
+
+        if (!`$mincpath/bin/minctoraw -v`) {
+            return new \LORIS\Http\Response\InternalServerError(
+                'minctoraw not installed'
+            );
+        }
+
+        $filename = $info->getFilename();
+        $fullpath = escapeshellarg($info->getPathname());
+
+        ob_start();
+        passthru("${mincpath}/bin/minctoraw -byte -unsigned -normalize $fullpath");
+        $content = ob_get_contents();
+        ob_end_clean();
```

Comment:
This would ideally be done using some kind of (theoretical) ExecStream or ProgramStream or something so that it could be streamed as it's running instead of holding the entire contents of the output in memory as a string, but that's not a blocker since it's far from the only inefficient place in LORIS right now..

Reply:
@driusan Would you care drafting something? I would be glad to add it.

Reply:
Maybe `\Zend\Diactoros\CallbackStream.php`